contracts/token/ERC721/ERC721.sol

```diff
@@ -308,7 +314,11 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata {
         // Clear approvals
         _approve(address(0), tokenId);
 
-        _balances[owner] -= 1;
+        unchecked {
+            // Cannot overflow, as that would require more tokens to be burned/transfered
+            // out then the owner initialy received through minting and transfering in.
+            _balances[owner] -= 1;
+        }
```

Comment:
For the same reason, because of `_beforeTokenTransfer` we can't use unchecked here.

```suggestion
        _balances[owner] -= 1;
```